HotelApp/Controllers/AccountController.cs

```diff
@@ -0,0 +1,118 @@
+            }
+            return Unauthorized();
+        }
+
+        [HttpPost]
+        [Route("register")]
+        public async Task<IActionResult> Register([FromBody] RegisterUserDTO model)
+        {
+            var userExists = await _userManager.FindByNameAsync(model.UserName);
+            if (userExists != null)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError,
+                                    new ResponseDTO
+                                    {
+                                        Status = "Error",
+                                        Message = "User already exists!"
+                                    });
+            }
+
+            User user = new User()
+            {
+                Email = model.Email,
+                SecurityStamp = Guid.NewGuid().ToString(),
+                UserName = model.UserName
+            };
+
+            var result = await _userManager.CreateAsync(user, model.Password);
+            if (!result.Succeeded)
+                return StatusCode(StatusCodes.Status500InternalServerError,
+                                    new ResponseDTO
+                                    {
+                                        Status = "Error",
+                                        Message = "Cannot create user! Please check registration details and try again."
+                                    });
+            return Ok(new ResponseDTO { Status = "Success", Message = "User created successfully!" });
+        }
+
+        private async Task<TokenResult> CreateTokenAsync(User user)
+        {
+            var claims = new List<Claim>();
+            claims.Add(new Claim(ClaimTypes.NameIdentifier, user.Id.ToString()));
+            claims.Add(new Claim(JwtRegisteredClaimNames.Sub, user.UserName));
+            claims.Add(new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString()));
+            var userRoles = await _userManager.GetRolesAsync(user);
+            claims.AddRange(userRoles.Select(x => new Claim(ClaimTypes.Role, x)));
+            var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_jwtSettings.Key));
+            var creds = new SigningCredentials(key, SecurityAlgorithms.HmacSha256);
+            var expires = DateTime.Now.AddDays(7);
+            var token = new JwtSecurityToken(
+                _jwtSettings.Issuer,
+                _jwtSettings.Audience,
+                claims,
+                expires: expires,
+                signingCredentials: creds
+            );
+            return new TokenResult
+            {
+                Token = new JwtSecurityTokenHandler().WriteToken(token),
+                Expiration = expires
+            };
+        }
+
+        [Authorize]
+        [HttpGet]
+        [Route("test")]
+        public async Task<IActionResult> Test()
+        {
+            return Ok(new ResponseDTO { Status = "Success", Message = "Neka test poruka" });
+        }
+    }
```

Comment:
Možeš obrisat ovu testnu metodu. To je samo da probaš lokalno, a na source ne stavljaš takve stvari